src/core/Akka.Persistence.TestKit.Tests/JournalInterceptorsSpecs.cs

```diff
@@ -67,20 +67,41 @@ public async Task cancelable_delay_must_call_next_interceptor_immediately_after_
             var totalDuration = 400.Milliseconds();
             var delayDuration = 200.Milliseconds();
             var epsilon = TimeSpan.FromMilliseconds(50);
+            
             using var cts = new CancellationTokenSource();
+            var synchronizationTcs = new TaskCompletionSource<bool>();
+            
+            // Custom interceptor that signals when it's called
             var probe = new InterceptorProbe();
+            probe.InterceptAsyncFunc = message =>
+            {
+                synchronizationTcs.TrySetResult(true);
+                return Task.CompletedTask;
+            };
+            
             var delay = new JournalInterceptors.CancelableDelay(totalDuration, probe, cts.Token);
 
             var startedAt = DateTime.Now;
             var task = delay.InterceptAsync(null);
+            
+            // Wait less than the full delay time
             await Task.Delay(delayDuration);
-
-            probe.WasCalled.Should().BeFalse();
+            
+            // Ensure the probe hasn't been called yet (not using probe.WasCalled since it might have race conditions)
+            synchronizationTcs.Task.IsCompleted.Should().BeFalse();
+            
+            // Cancel the delay
             cts.Cancel();
-            await task;
-
+            
+            // Wait for the probe to be called
+            await synchronizationTcs.Task;
```

Comment:
Adds a more deterministic method for making sure the probe gets called before we assert